### PR TITLE
fix dependency order when package publish

### PR DESCRIPTION
**Pull request by baichuan3**
An error will be reported when deploying multiple contracts, resolve https://github.com/rooch-network/rooch/issues/116

`
VMError with status LINKER_ERROR at location UNDEFINED and message Cannot find ModuleId { address: 0000000000000000000000000000000000000000000000000000000000000123, name: Identifier("something_do_logic") } in data cache

`

Ω

### CODE REVIEW

File: moveos/moveos/src/vm/dependency_order.rs
Comment (jolestar) [inline moveos/moveos/src/vm/dependency_order.rs:1]: I think we can directly use the Move's `dependency_order.rs`, you can see the Framework build.

Comment (jolestar) [reply moveos/moveos/src/vm/dependency_order.rs:1]: The framework build package all dependency into one package. For DApp, We just package the project's module, call `package.root_modules_map()`

https://github.com/rooch-network/rooch/blob/78a263912d0def756a27c4567d7b70c6a979f98a/moveos/moveos-stdlib/src/lib.rs#L81-L89

Comment (baichuan3) [reply moveos/moveos/src/vm/dependency_order.rs:1]: yes, use package the project's module, call package.root_modules_map()

you can use this branch for test ,  https://github.com/rooch-network/rooch/tree/bugfix_publish_order_tmp

Step1:
` rooch server start`

Step2:
`rooch move publish -p examples/basic_object_example --sender-account 0x123`

then occur error,
`thread 'main' panicked at 'Missing dependency 000000000000000000000000000000000000000000000000000000000000001::object',~/.cargo /git/checkouts/move-d59e49550a684528 /5b413d0/language/tools/move-bytecode-utils/src/dependency_graph.rs:46:44`

call entry is here, https://github.com/rooch-network/rooch/blob/5b11359c4844dbe825178b843792e4f344de854a/moveos/moveos-cli/src/commands/publish.rs#L49

**Comment by jolestar** [comment]
Maybe this is a bug of Move's `dependency_order.rs`. We can merge first, then try to fix it on upstream. #124

**Review by jolestar — APPROVED** [review]

**Pull request merged**